### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by ajs6f**
Just removing dead code and variables that don't get used, adding `static` here or there to clarify tests a bit, that sort of thing. Pruning.

### CODE REVIEW

**Comment by ajs6f** [comment]
A few more commits to come this afternoon.

**Comment by afs** [comment]
Can this wait until after the 3.1.1 release?

**Comment by ajs6f** [comment]
Absolutely. Would you rather I just left it here and we can pick it up at the right moment, or close this PR and reopen it post-release?

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java [hunk 1]
```diff
@@ -371,19 +367,6 @@ else if ( ! dir.isDirectory())
             throw new FusekiConfigException("Not a directory: "+directory) ;
     }
 
-    private static void mustExist(Path directory) {
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: Please leave. While unused, these 3 methods form verbs for working with Paths - I'd prefer to leave the verbs around.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public AsyncTask submit(Runnable task, String displayName, DataService dataServi
             Fuseki.serverLog.info(format("Task : %s : %s",taskId, displayName)) ;
             Callable<Object> c = Executors.callable(task) ;
             AsyncTask asyncTask = new AsyncTask(c, this, taskId, displayName, dataService, requestId) ;
-            Future<Object> x = executor.submit(asyncTask) ;
+            executor.submit(asyncTask) ;
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java]: I prefer the assignment.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java [hunk 3]
```diff
@@ -162,10 +162,6 @@
         }
     }
     
-    private static boolean emptyDir(Path dir) {
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: Move down to near `mustExist`

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java [hunk 4]
```diff
@@ -371,19 +367,6 @@ else if ( ! dir.isDirectory())
...
 
-    private static void mustExist(Path directory) {
-        File dir = directory.toFile() ;
-        if ( ! dir.exists() )
-            throw new FusekiConfigException("Does not exist: "+directory) ; 
-        if ( ! dir.isDirectory())
-            throw new FusekiConfigException("Not a directory: "+directory) ;
-    }
-    
-    private static boolean exists(Path directory) {
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: 3rd of 3

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/validation/DataValidator.java [hunk 5]
```diff
@@ -101,33 +95,6 @@ protected JsonObject execute(ValidationAction action) {
     protected String validatorName() {
         return "RDF Data" ;
     }
-    
-    // Error handler that records messages
-    private static class ErrorHandlerMsg implements ErrorHandler
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/validation/DataValidator.java]: Please leave.

File: jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/FileSender.java [hunk 6]
```diff
@@ -79,8 +79,8 @@ public void send(String method) {
                 }
                 ps.print("--" + boundary + "--"+WNL);
             } 
-            connection.connect(); 
-            int responseCode = connection.getResponseCode();
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/FileSender.java]: I prefer the assignment - helps debugging!

File: jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/TestAdmin.java [hunk 7]
```diff
@@ -309,11 +310,6 @@
         }
     }
 
-    private JsonValue getTask(String taskId) {
-        String url = urlRoot+"$/tasks/"+taskId ;
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/TestAdmin.java]: A useful verb (reflect lack of code for batch task handling).

File: jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/TestQuery.java [hunk 8]
```diff
@@ -261,18 +260,10 @@ public void query_describe_conneg() {
...
-    private void execQuery(String queryString, int exceptedRowCount) {
+    private static void execQuery(String queryString, int exceptedRowCount) {
         QueryExecution qExec = QueryExecutionFactory.sparqlService(serviceQuery, queryString) ;
         ResultSet rs = qExec.execSelect() ;
         int x = ResultSetFormatter.consume(rs) ;
         assertEquals(exceptedRowCount, x) ;
     }
-
-    private void execQuery(String queryString, ResultSet expectedResultSet) {
-        QueryExecution qExec = QueryExecutionFactory.sparqlService(serviceQuery, queryString) ;
```
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/test/java/org/apache/jena/fuseki/TestQuery.java]: An action that supports the tests - even if not used at the moment, it shows what is possible.

**Comment by ajs6f** [comment]
I will try to fix these things sometime tomorrow in another commit, especially if I can fix #151.

**Comment by afs** [comment]
I didn't review jena-permissions or jena-jdbc-* or jena-spatial.

File: jena-base/src/main/java/org/apache/jena/atlas/io/BlockUTF8.java [hunk 9]
```diff
@@ -154,11 +152,6 @@ else if ( (x & 0xF8) == 0xF0 )
             }
         }
     }
-
-    private static void toCharsArray(byte[] bytes, char[] chars)
-    {
-        throw new NotImplemented() ;
-    }
```
Comment (ajs6f) [inline jena-base/src/main/java/org/apache/jena/atlas/io/BlockUTF8.java]: Will do.

File: jena-base/src/main/java/org/apache/jena/atlas/io/BlockUTF8.java [hunk 10]
```diff
@@ -227,11 +220,6 @@ else if ( ch <= 0x7FFFFFFF )
         }
         //int bytesFinish = bb.position() ;
     }
-
-    private static void fromCharsArray(char[] array, byte[] array2)
-    {
-        throw new NotImplemented() ;
-    }
```
Comment (ajs6f) [inline jena-base/src/main/java/org/apache/jena/atlas/io/BlockUTF8.java]: Will do.

File: jena-base/src/test/java/org/apache/jena/atlas/lib/TestHex.java [hunk 11]
```diff
@@ -85,7 +85,7 @@ private static void testStr2Val(String str, int expected) {
     }
     
     private static void testStr2Val(String str, int start, int length, int expected) {
-        int v = Hex.hexStringToInt(str, start, length) ;
+        Hex.hexStringToInt(str, start, length) ;
```
Comment (ajs6f) [inline jena-base/src/test/java/org/apache/jena/atlas/lib/TestHex.java]: Okay.

File: jena-base/src/test/java/org/apache/jena/atlas/io/AbstractTestPeekInputStream.java [hunk 12]
```diff
@@ -140,7 +139,7 @@
     {
         PeekInputStream in = make("abc") ;
         checkLineCol(in, INIT_LINE, INIT_COL) ;
-        int ch = in.readByte() ;
+        in.readByte() ;
```
Comment (ajs6f) [inline jena-base/src/test/java/org/apache/jena/atlas/io/AbstractTestPeekInputStream.java]: Okay.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java (hunk 1 above)
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: Okay.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java (hunk 3 above)
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: Okay.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java (hunk 4 above)
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/FusekiServer.java]: All in a group at the end.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java (hunk 2 above)
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java]: Why?

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/validation/DataValidator.java (hunk 5 above)
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/validation/DataValidator.java]: Okay-- maybe we could at least exercise it in a test?

**Comment by ajs6f** [comment]
The bulk of changes to `jena-permissions` are using `<>` instead of writing out type parameters, but there are also a few more substantive syntax and other changes there.

File: jena-permissions/src/main/java/org/apache/jena/permissions/impl/SecuredItemImpl.java [hunk 13]
```diff
@@ -299,7 +299,7 @@ public String toString() throws AuthenticationRequiredException {
 	 * @return the value of the security check or <code>null</code> if the value
 	 *         has not been cached.
 	 */
-	private Boolean cacheGet(final CacheKey key) {
+	private static Boolean cacheGet(final CacheKey key) {
```
Comment (Claudenw) [inline jena-permissions/src/main/java/org/apache/jena/permissions/impl/SecuredItemImpl.java]: What is the reasoning for making this a static method?  While it is using thread local only it seems to me that this is not semantically the same as a static method.

File: jena-permissions/src/main/java/org/apache/jena/permissions/impl/SecuredItemImpl.java [hunk 14]
```diff
@@ -312,7 +312,7 @@ private Boolean cacheGet(final CacheKey key) {
 	 * @param value
 	 *            The value to set.
 	 */
-	private void cachePut(final CacheKey key, final boolean value) {
+	private static void cachePut(final CacheKey key, final boolean value) {
```
Comment (Claudenw) [inline jena-permissions/src/main/java/org/apache/jena/permissions/impl/SecuredItemImpl.java]: Again, this change modifies the is not semantic meaning.

Comment (ajs6f) [inline jena-permissions/src/main/java/org/apache/jena/permissions/impl/SecuredItemImpl.java]: See earlier comment.

**Comment by ajs6f** [comment]
> Unless there is a good reason to make the 2 permissions methods static I would prefer to leave them alone.

I had some reason (explained above) but since seems less clear to you, I will remove it.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java (hunk 2 above)
Comment (afs) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java]: It returns a result.  I prefer to assign it and not leave it hidden.

Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java]: Done.

**Comment by ajs6f** [comment]
I think I addressed the comments given here in that last commit.

File: jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/mgt/ActionDatasets.java [hunk 15]
```diff
@@ -111,8 +110,6 @@ protected JsonValue execGetItem(HttpAction action) {
     @Override
     protected JsonValue execPostContainer(HttpAction action) {
         JenaUUID uuid = JenaUUID.generate() ;
```
Comment (ajs6f) [inline jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/mgt/ActionDatasets.java]: It seems to be used at 137?

**Comment by ajs6f** [comment]
Glad to!

**Comment by ajs6f** [comment]
@Claudenw are you okay with this PR now?

**Comment by ajs6f** [comment]
@Claudenw If you have no further objection, I'm going to merge this EOD today.

**Comment by ajs6f** [comment]
@Claudenw @afs Hang on-- I just reread that @afs would like this to hold until post-3.1.1, so I won't merge it after all.

**Comment by afs** [comment]
Given the delay due to sorting out HttpOp etc, it can go in from my POV.

**Comment by ajs6f** [comment]
Oh, cool. Okay, back to "will merge EOD today unless @Claudenw objects".

**Pull request merged**